optuna/integration/allennlp/_executor.py

```diff
@@ -212,12 +213,30 @@ def run(self) -> float:
 
         self._set_environment_variables()
         params = allennlp.common.params.Params(self._build_params())
-        allennlp.commands.train.train_model(
-            params=params,
-            serialization_dir=self._serialization_dir,
-            file_friendly_logging=self._file_friendly_logging,
-            force=self._force,
-            include_package=self._include_package,
-        )
+
+        if "distributed" in params:
+
+            if OPTUNA_ALLENNLP_DISTRIBUTED_FLAG in os.environ:
+                warnings.warn(
+                    "Other process may already exists."
+                    " If you have trouble, please unset the environment"
+                    " variable `OPTUNA_ALLENNLP_USE_DISTRIBUTED`"
+                    " and try it again."
+                )
+
+            os.environ[OPTUNA_ALLENNLP_DISTRIBUTED_FLAG] = "1"
+
+        try:
+            allennlp.commands.train.train_model(
+                params=params,
+                serialization_dir=self._serialization_dir,
+                file_friendly_logging=self._file_friendly_logging,
+                force=self._force,
+                include_package=self._include_package,
+            )
+        except ProcessRaisedException as e:
+            if "raise TrialPruned()" in str(e):
```

Comment:
Error massage looks like following:

```
-- Process 0 terminated with the following error:
Traceback (most recent call last):
  File "/path/to/optuna/venv/lib/python3.9/site-packages/torch/multiprocessing/spawn.py", line 59, in _wrap
    fn(i, *args)
  File "/path/to/optuna/venv/lib/python3.9/site-packages/allennlp/commands/train.py", line 504, in _train_worker
    metrics = train_loop.run()
  File "/path/to/optuna/venv/lib/python3.9/site-packages/allennlp/commands/train.py", line 577, in run
    return self.trainer.train()
  File "/path/to/optuna/venv/lib/python3.9/site-packages/allennlp/training/gradient_descent_trainer.py", line 750, in train
    metrics, epoch = self._try_train()
  File "/path/to/optuna/venv/lib/python3.9/site-packages/allennlp/training/gradient_descent_trainer.py", line 857, in _try_train
    callback.on_epoch(self, metrics=metrics, epoch=epoch, is_primary=self._primary)
  File "/path/to/optuna/optuna/integration/allennlp/_pruner.py", line 250, in on_epoch
    raise TrialPruned()
optuna.exceptions.TrialPruned
```

I inspect message and check if `"raise` TrialPruned()"` appears in it.